worker/dbaccessor/repl.go

```diff
@@ -109,14 +109,22 @@ func (r *sqlREPL) registerCommands() {
 }
 
 func (r *sqlREPL) acceptConnections() {
-	defer r.sessionGroup.Done()
+	// If we're not listening, don't attempt to close the wait group.
```

Comment:
Why not? The code should work as-is whether we are listening or not. Am I missing something here?